terraform/addons/monitoring/lambda/main.go

```diff
@@ -0,0 +1,190 @@
+
+    - cron_stats
+
+If we have an old/incomplete run in cron_stats or if we are missing a
+cron entry entirely, throw an alert to an SNS topic.
+
+Currently tested crons:
+
+    - cleanups_then_aggregation
+    - vulnerabilities
+
+*/
+
+package main
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/aws/aws-secretsmanager-caching-go/secretcache"
+	"github.com/go-sql-driver/mysql"
+	flags "github.com/jessevdk/go-flags"
+)
+
+type NullEvent struct{}
+
+type OptionsStruct struct {
+	LambdaRuntimeAPI   string `long:"lambda-runtime-api" env:"AWS_LAMBDA_RUNTIME_API"`
+	SNSTopicArns       string `long:"sns-topic-arn" env:"SNS_TOPIC_ARNS" required:"true"`
+	MySQLHost          string `long:"mysql-host" env:"MYSQL_HOST" required:"true"`
+	MySQLUser          string `long:"mysql-user" env:"MYSQL_USER" required:"true"`
+	MySQLSMSecret      string `long:"mysql-secretsmanager-secret" env:"MYSQL_SECRETSMANAGER_SECRET" required:"true"`
+	MySQLDatabase      string `long:"mysql-database" env:"MYSQL_DATABASE" required:"true"`
+	FleetEnv           string `long:"fleet-environment" env:"FLEET_ENV" required:"true"`
+	AWSRegion          string `long:"aws-region" env:"AWS_REGION" required:"true"`
+	CronDelayTolerance string `long:"cron-delay-tolerance" env:"CRON_DELAY_TOLERANCE" default:"2h"`
+}
+
+var options = OptionsStruct{}
+
+func sendSNSMessage(msg string, sess *session.Session) {
+	log.Printf("Sending SNS Message")
+	fullMsg := fmt.Sprintf("Environment: %s\nMessage: %s", options.FleetEnv, msg)
+	svc := sns.New(sess)
+	for _, SNSTopicArn := range strings.Split(options.SNSTopicArns, ",") {
+		log.Printf("Sending '%s' to '%s'", fullMsg, SNSTopicArn)
+		result, err := svc.Publish(&sns.PublishInput{
+			Message:  &fullMsg,
+			TopicArn: &SNSTopicArn,
+		})
+		if err != nil {
+			log.Printf(err.Error())
+		}
+		log.Printf(result.GoString())
+	}
+}
+
+func checkDB(sess *session.Session) (err error) {
+	secretCache, err := secretcache.New()
+	if err != nil {
+		log.Printf(err.Error())
+		sendSNSMessage("Unable to initialise SecretsManager helper.  Cron status is unknown.", sess)
+		return err
+	}
+
+	secretCache.Client = secretsmanager.New(sess)
+
+	MySQLPassword, err := secretCache.GetSecretString(options.MySQLSMSecret)
+	if err != nil {
+		log.Printf(err.Error())
+		sendSNSMessage("Unable to retrieve SecretsManager secret.  Cron status is unknown.", sess)
+		return err
+	}
+
+	cfg := mysql.Config{
+		User:                 options.MySQLUser,
+		Passwd:               MySQLPassword,
+		Net:                  "tcp",
+		Addr:                 options.MySQLHost,
+		DBName:               options.MySQLDatabase,
+		AllowNativePasswords: true,
+		ParseTime:            true,
+	}
+
+	db, err := sql.Open("mysql", cfg.FormatDSN())
```

Comment:
necessary to add `defer db.Close()`  as to not leak db connections between lambda invocations. I assume that handle() might be called multiple times for the same provisioned runtime.